backend/main.py

```diff
@@ -63,31 +90,34 @@ async def websocket_endpoint(websocket: WebSocket):
     
     await manager.connect(websocket)
 
-    # Maybe make this "while True" a "while connected"?
-    while True:
-        current_players = game_handler.players
-        
-        data = await websocket.receive_text()
-        message = json.loads(data)
-        
-        if message["type"] == "join":
-            name = message["name"]
-            game_handler.join(websocket,name)
-
-            for ws in current_players:
-                await ws.send_text(f"{name} joined the lobby!")
+    try :
+        # Maybe make this "while True" a "while connected"?
+        while True:
+            current_players = game_handler.players
             
-        elif message["type"] == "bid":
-            if not game.ongoing:
-                await websocket.send_text(f"A game is not currently in progress.")
-            else:
-                amount = message["amount"]
-                await manager.broadcast(f"{current_players[ws]} bid {amount}.")
-        
-        elif message["type"] == "cashout":
-            if not game.ongoing:
-                await websocket.send_text(f"A game is not currently in progress.")
-            else:
-                await manager.broadcast(f"{current_players[ws]} cashed out with a mult of {game.get_multiplicator()}!")
-    
-    await manager.disconnect(websocket)
+            data = await websocket.receive_text()
+            message = json.loads(data)
+            
+            if message["type"] == "join":
+                name = message["name"]
+                game_handler.join(websocket,name)
+                await manager.broadcast_lobby(f'{name} joined the lobby!')
+                # A player has joined
+                player_join_event.set() 
+                
+            elif message["type"] == "bid":
+                if not game.ongoing:
+                    await manager.send_personal_message_lobby(f"A game is not currently in progress.")
+                else:
+                    amount = message["amount"]
+                    await manager.broadcast_lobby(f"{current_players[websocket]} bid {amount}.")
+            
+            elif message["type"] == "cashout":
+                if not game.ongoing:
+                    await manager.send_personal_message_lobby(f"A game is not currently in progress.")
+                else:
+                    await manager.broadcast_lobby(f"{current_players[websocket]} cashed out with a mult of {game.get_multiplicator()}!")
```

Comment:
Sorry Theo for complaining again about this.
But the messages need to be "structured", in the sense they need to be understood by the frontend so the events can be rendered.

For example, a join event might look something like (just a suggestion):
```
{
  "type": "join",
  "target": "Theo",
  "lobby": ["Nicolas", "Theo"]
}
```
(btw on a sidenote - I realize the `join` event is one that goes both client -> server and server -> client and has different meanings. On the `client -> server`, it's a "I'm here to join the lobby". On the `server -> client`, it's a "hey guys, this new person has just joined".

This message is parsed by the frontend, which then updates the lobby to render the updated state, perhaps printing a message saying `Theo joined`